cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -0,0 +1,595 @@
+            flushToStorage(debugStreamSegmentContainer);
+            attempts++;
+        }
+        output("Reconciling container and storage metadata segments");
+        reconcileStorageSegment(debugStreamSegmentContainer);
+        overrideLogMetadataWithEpoch(containersToEpoch, dataLogFactory);
+        output("----------------Stopping DebugSegmentContainer %d----------------", containerId);
+        debugStreamSegmentContainer.close();
+        Thread.sleep(5000);
+    }
+
+    /**
+     * Overrides the given DurableDataLog metadata with provided epoch.
+     * @param containersToEpoch Map of all containers to their highest epoch.
+     * @param durableDataLogFactory Factory used to create DurabelDataLog
+     */
+    @SneakyThrows
+    private void overrideLogMetadataWithEpoch(Map<Integer, Long> containersToEpoch, DurableDataLogFactory durableDataLogFactory) {
+        for (Map.Entry<Integer, Long> entry: containersToEpoch.entrySet()) {
+            DebugDurableDataLogWrapper debugDurableDataLogWrapper = durableDataLogFactory.createDebugLogWrapper(entry.getKey());
+            debugDurableDataLogWrapper.overrideEpochInMetadata(entry.getValue());
+        }
+    }
+
+    /**
+     * Method to set the highest epoch seen while parsing metadata chunks.
+     * @param chunk Chunk to extract epoch from.
+     * @param containerId Container ID of the the container being recovered.
+     * @param containersToEpoch Map to hold highest epoch for all containers.
+     */
+    private void setEpochforContainer(File chunk, int containerId, Map<Integer, Long> containersToEpoch) {
+        long epoch = getEpochFromChunk(chunk);
+        long epochAlreadyPresent = containersToEpoch.getOrDefault(containerId, Long.valueOf(1));
+        if ( epoch > epochAlreadyPresent ) {
+            containersToEpoch.put(containerId, epoch);
+        }
+    }
+
+    /**
+     * Extracts epoch from chunk name.
+     */
+    private long getEpochFromChunk(File chunk) {
+        String[] chunkParts = chunk.getName().split(EPOCH_SPLITTER);
+        return Long.parseLong(chunkParts[chunkParts.length - 1].split("-")[0]);
+    }
+
+    private File[] getContainerMetadataChunkFiles(File[] allFiles, int containerId) {
+        List<File> filteredFiles =  Arrays.stream(allFiles)
+                .filter(File::isFile)
+                .filter(f -> !f.getName().contains(ATTRIBUTE_SUFFIX))
+                .filter(f -> f.getName().split(EPOCH_SPLITTER)[0].contains("_" + String.valueOf(containerId)))
+                .collect(Collectors.toList());
+        return filteredFiles.toArray(new File[filteredFiles.size()]);
+    }
+
+    private void deleteJournalsOf(int containerId) {
+        File dir = new File(tier2Root + containersPath);
+        assert dir != null;
+        File[] files = dir.listFiles();
+        assert files != null;
+        for (File file : files) {
+            if (file.getName().contains("container" + String.valueOf(containerId))) {
```

Comment:
I think we should have a stricter name match so that we don't accidently/unintentionally delete unrelated file.
see https://github.com/pravega/pravega/blob/0ce0d8528c7df1ff9172382331629d2606732746/shared/protocol/src/main/java/io/pravega/shared/NameUtils.java#L195-L203

Reply:
made it stricter by choosing all files that start with "_sysjournal". Also changed the functionality to backing up journal files instead of deleting them.